### PR TITLE
Standalone CLI

### PR DESCRIPTION
**Pull request by jenow**
## What does this PR do?

Externalize the Kuzzle CLI here.

Make `encryptSecrets` and `decryptSecrets` using the [kuzzle-vault](https://github.com/kuzzleio/kuzzle-vault) module.

~~:warning: Depends on https://github.com/kuzzleio/kuzzle-vault/pull/1 to be released on npm! Once it will be released we have to change the dependecy in the package.json :warning:~~

See [Readme](https://github.com/kuzzleio/kuzzle-cli/blob/92af42cfcfb57569f17ebea4aa0b9d5ef3578761/README.md) for more

## How should this be manually tested?

```sh
npm run test
```

### CODE REVIEW

File: commands/encryptSecrets.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  if (!userIsSure) {
+    cout.notice('[ℹ] Aborted');
+    process.exit(1);
+  }
+
+  cout.notice('[ℹ] Encrypting secrets...\n');
+
+  let sdk;
+
+  getSdk(options, 'ws')
```
Comment (Aschen) [inline commands/encryptSecrets.js]: You should display a warning message with confirmation if an user try to encrypt/decrypt secrets without SSL on a remote Kuzzle server

**Review by Aschen — CHANGES_REQUESTED** [review]
~We should add an option to instantiate the protocol with SSL support before sending the secrets to Kuzzle but I think this can be done in a separate PR.~
Obsolete since vault externalization

File: .ci/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      - redis
+      - elasticsearch
+    environment:
+      - kuzzle_services__db__client__host=http://elasticsearch:9200
+      - kuzzle_services__internalCache__node__host=redis
+      - kuzzle_services__memoryStorage__node__host=redis
+      - NODE_ENV=production
+
+  redis:
+    image: redis:3.2
```
Comment (Aschen) [inline .ci/docker-compose.yml]:
```suggestion
    image: redis:5
```

**Review by Aschen — COMMENTED** [review]
It would have been better to use JS SDK in functional tests but I know writing tests is a pain in the ass so is up to you

**Comment by jenow** [comment]
> It would have been better to use JS SDK in functional tests but I know writing tests is a pain in the ass so is up to you

I only took the tests from Kuzzle and put them here (also make them use async / await). We could use the sdk to write the missing tests after this PR.

**Review by Aschen — APPROVED** [review]

**Review by alexandrebouthinon — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+## Install dependencies
+
+Simply run
+
+```sh
+npm install
+```
+
+## Available commands
```
Comment (xbill82) [inline README.md:13]: Lacks of a summary like 
`kuzzle-cli <command> [options]`

File: README.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+### createFirstAdmin
```
Comment (xbill82) [inline README.md]: Would be great that these titles were formatted like code
```suggestion
### `createFirstAdmin`
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+    --noint : non interactive mode
+
+### decryptSecrets [file]
+
+Decrypt a secrets file with the provided key.
+
+Options:
+
+    --vault-key <vaultKey> : Vault key used to decrypt secrets
```
Comment (xbill82) [inline README.md]:
```suggestion
    `--vault-key <vaultKey>` : Vault key used to decrypt secrets
```

Comment (xbill82) [reply README.md]: Same with the other ones

File: build-cli.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+ * limitations under the License.
+ */
+
+'use strict';
+
+const
+  semver = require('semver'),
+  { execSync } = require('child_process');
+
+if (!semver.satisfies(process.version, '>= 8.0.0')) {
```
Comment (xbill82) [inline build-cli.js:30]: Are we sure this is correct? I'd rather say use Babel if the current Node version is _less than_ 8.

File: commands/clearCache.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+function commandClearCache (database, options) {
+  let
+    opts = options,
+    db = database,
+    userIsSure;
+
+  if (options === undefined) {
+    opts = database;
+    db = 'internalCache';
```
Comment (xbill82) [inline commands/clearCache.js:37]: Shouldn't this be conditioned to `database` having falsey value?

File: commands/indexDump.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    if (! stream.write(data)) {
+      stream.once('drain', resolve);
+    }
+    else {
+      resolve();
+    }
+  });
+}
+
+async function dumpCollection (sdk, index, collection, batchSize, directoryPath) {
```
Comment (xbill82) [inline commands/indexDump.js:55]: Why are we using `async` here? We use promise chains in all the previous files...

Comment (Aschen) [reply commands/indexDump.js:55]: Writing this function without async/await is really a mess, this is the reason why I added babel to the CLI

File: .eslintrc [hunk 9]
```diff
@@ -0,0 +1,64 @@
+{
```
Comment (scottinet) [inline .eslintrc]: This file should be removed: duplicate of .eslintrc.json

**Review by scottinet — CHANGES_REQUESTED** [review]

File: build-cli.js (hunk 6 above)
Comment (Aschen) [reply build-cli.js:30]: The condition is inversed

**Pull request merged**